projects/concentrator/index.js

```diff
@@ -29,7 +31,9 @@ const replacements = [
 
 async function tvl(timestamp, block) {
   let balances = {}
-
+  const initInfo = await fetchURL('http://concentrator-api.aladdin.club/data/aladdin/initInfo');
```

Comment:
hi, can you compute this using on-chain calls instead?

Reply:
> hi, can you compute this using on-chain calls instead?

code is complex

Reply:
@jianfengxiaojianfeng can you explan what is happening a bit, I can try to convert it on chain.

Reply:
compute TVL on-chain

Reply:
@jianfengxiaojianfeng thanks for turning it onchain